python/stempy/io/sparse_array.py

```diff
@@ -188,19 +190,31 @@ def from_hdf5(cls, filepath, **init_kwargs):
 
             frames = f['electron_events/frames']
             scan_positions_group = f['electron_events/scan_positions']
-
-            data = frames[()]
             scan_shape = [scan_positions_group.attrs[x] for x in ['Nx', 'Ny']]
             frame_shape = [frames.attrs[x] for x in ['Nx', 'Ny']]
-
-            scan_positions = scan_positions_group[()]
+            
+            if keep_flyback:
+                data = frames[()] # load the full data set
+                scan_positions = scan_positions_group[()]
+            else:
+                # Generate the original scan indices from the scan_shape
+                orig_indices = np.ravel_multi_index([ii.ravel() for ii in np.indices(scan_shape)],scan_shape)
```

Comment:
any reason to use `ii` for indices? instead of `i` for ex.

Reply:
I never use single letter variables.

I once spent a full day debugging something where `i` was already used for `sqrt(-1)`. Ever since, I always use `ii`, `jj`, `kk`, etc. for loops.

Also, when using the `pdb` [debugger](https://docs.python.org/3/library/pdb.html#pdbcommand-next), `c`, `q`, `n`, etc. are reserved for interacting with the debugger (continue, quit, next, etc.). That messed me up a few times as well.